### ISSUE TITLE
Typescript: .first('*') returns `any[]`

### ISSUE THREAD

**Issue by DataTables** [knex/knex#3619]
# Environment

Knex version: 0.20.4
Database + version: Postgres 10
OS: Fedora

@lorefnon 

# Bug

1) Explain what kind of behaviour you are getting and how you think it should do

This is a continuation of #3229 which noted that `first()` was returning `any[]`. That has been fixed if you don't specify any columns, but if you do `.first('*')` the return type is still `any[]`.

2) Error message

Typescript error message:

> Property 'whatever' does not exist on type 'any[]'.

3) Reduced test code, for example in https://npm.runkit.com/knex or if it needs real
   database connection to MySQL or PostgreSQL, then single file example which initializes
   needed data and demonstrates the problem.

```
let result = await knex
  .first('*')
  .from('table')
  .where({
    whatever: whatever
});

result.id // typescript error
```

**Comment by lorefnon** [comment]
Thanks for reporting this. 

On an unrelated note, thanks for creating DataTables :). We used it for multiple years in several projects.

**Comment by kibertoad** [comment]
Released in 0.20.8
